aws/sdk-codegen/src/main/kotlin/software/amazon/smithy/rustsdk/SigV4SigningDecorator.kt

```diff
@@ -78,20 +80,40 @@ class SigV4SigningConfig(private val sigV4Trait: SigV4Trait) : ConfigCustomizati
     }
 }
 
-class SigV4SigningFeature(private val runtimeConfig: RuntimeConfig) :
+fun needsAmzSha256(service: ServiceShape) = when {
+    service.id == ShapeId.from("com.amazonaws.s3#AmazonS3") -> true
```

Comment:
There's nothing in the Smithy model that tells us this?

Reply:
nope signing is the wild west: https://awslabs.github.io/smithy/1.0/spec/aws/aws-auth.html#aws-auth-sigv4-trait